features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -2979,9 +2981,10 @@ ble_error_t GenericGap::startScan(
         );
 
         if (err) {
-            _scan_enabled = false;
             return err;
         }
+
+        _scan_enabled = true;
```

Comment:
`_scan_enabled = true` can be moved out the if/else branches; right before the return statement.

Reply:
moved